packages/app/src/app/location-selection/polygon-map.service.ts

```diff
@@ -0,0 +1,430 @@
+import { BehaviorSubject, finalize, Subject, tap } from 'rxjs';
+import { WebApiService } from '../../api/web-api.service';
+import { LayerProperties } from '../../types/layer.type';
+import { disposeLayerGroup } from '../map/openlayers-util';
+
+export const USER_POLYGON_LAYER_ID = 'user-polygon-layer';
+
+/**
+ * Service for managing user-drawn polygons on the map.
+ * Handles fetching polygons from the API and rendering them as OpenLayers vector layers.
+ */
+@Injectable()
+export class PolygonMapService {
+  private readonly destroyRef = inject(DestroyRef);
+  private readonly injector = inject(Injector);
+  private readonly api = inject(WebApiService);
+  private readonly snackbar = inject(MatSnackBar);
+
+  private map!: OLMap;
+  private readonly geojsonFormat = new GeoJSON();
+
+  /**
+   * Layer group containing all user polygon layers
+   */
+  private readonly polygonLayerGroup = signal<LayerGroup | undefined>(undefined);
+
+  /**
+   * Currently loaded polygons (by ID for efficient lookup)
+   */
+  private readonly polygonsById = new Map<number, PolygonReference>();
+
+  /**
+   * Map of polygon ID to OpenLayers feature for efficient updates
+   */
+  private readonly featuresByPolygonId = new Map<number, Feature<Geometry>>();
+
+  /**
+   * Currently loaded polygons array
+   */
+  private readonly polygons = signal<PolygonReference[]>([]);
+
+  /**
+   * Loading state
+   */
+  readonly loading = signal(false);
+
+  /**
+   * Trigger to refresh polygons from API
+   */
+  private readonly refreshTrigger$ = new Subject<{ projectId: number }>();
+
+  /**
+   * Observable of current polygons
+   */
+  readonly polygons$ = new BehaviorSubject<PolygonReference[]>([]);
+
+  /**
+   * Currently active project ID (for tracking state)
+   */
+  private currentProjectId?: number;
+
+  constructor() {
+    this.setupRefreshListener();
+  }
+
+  /**
+   * Set the map instance
+   * @param map OpenLayers map
+   * @param projectId Optional project ID to load polygons for
+   */
+  configureMapService(map: OLMap, projectId: number): void {
+    this.map = map;
+    this.currentProjectId = projectId;
+
+    // Initial load of polygons
+    this.refresh(projectId);
+  }
+
+  /**
+   * Setup listener for refresh triggers
+   */
+  private setupRefreshListener(): void {
+    this.refreshTrigger$.pipe(takeUntilDestroyed(this.destroyRef)).subscribe(({ projectId }) => {
+      this.fetchAndRenderPolygons(projectId);
+    });
+  }
+
+  /**
+   * Trigger a refresh of polygons from the API
+   * @param projectId project ID to filter polygons by
+   */
+  refresh(projectId: number): void {
+    this.currentProjectId = projectId;
+    this.refreshTrigger$.next({ projectId });
+  }
+
+  /**
+   * Fetch polygons from API and render them on the map
+   * @param projectId project ID to filter polygons by
+   */
+  private fetchAndRenderPolygons(projectId: number): void {
+    if (!this.map) {
+      console.warn('Map not initialized, cannot fetch polygons');
```

Comment:
technically can't render them. if wanted to could build up a rxjs chain that waits on map, projectId, which would be more robust if you have race conditions. but if this is working don't worry about it